lib/container_server.go

```diff
@@ -53,7 +54,41 @@ func (c *ContainerServer) Runtime() *oci.Runtime {
 
 // Hooks returns the oci hooks for the ContainerServer
 func (c *ContainerServer) Hooks() map[string]HookParams {
-	return c.hooks
+	hooks := map[string]HookParams{}
+	c.hooksLock.Lock()
+	defer c.hooksLock.Unlock()
+	for key, h := range c.hooks {
+		hooks[key] = h
+	}
+	return hooks
+}
+
+// RemoveHook removes an hook by name
+func (c *ContainerServer) RemoveHook(hook string) {
+	c.hooksLock.Lock()
+	defer c.hooksLock.Unlock()
+	if _, ok := c.hooks[hook]; ok {
+		delete(c.hooks, hook)
+	}
+}
+
+// AddHook adds an hook by hook's path
+func (c *ContainerServer) AddHook(hookPath string) {
+	c.hooksLock.Lock()
+	defer c.hooksLock.Unlock()
+	hook, err := readHook(hookPath)
+	if err != nil {
+		logrus.Debugf("error while reading hook %s", hookPath)
+		return
+	}
+	for key, h := range c.hooks {
+		// hook.Hook can only be defined in one hook file, unless it has the
+		// same name in the override path.
+		if hook.Hook == h.Hook && key != filepath.Base(hookPath) {
```

Comment:
I don't see a reason that `hook.Hook` should be specific to one hook file.  For example, you may want to limit a hook to one stage for an `annotations` match and another stage for a `cmd` match.  Using two files with the same `hook` entry is the only way to do that.  I think we can drop the `hook.Hook == h.Hook &&` from this comparison and only look at the filename (since that's the key used for clobbering in `c.hooks`).